### PR TITLE
Fixes #14113 - run dhcp orchestration only on DHCP subnets

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by dLobatog** [comment]
[test]

**Comment by dLobatog** [comment]
Looking good, will merge if tests pass

File: test/unit/orchestration/dhcp_test.rb [hunk 1]
```diff
@@ -53,6 +53,17 @@ def teardown
     end
   end
 
+  test 'bmc_should_have_valid_dhcp_record' do
```
Comment (domcleal) [inline test/unit/orchestration/dhcp_test.rb]: This should be changed too.

Comment (ares) [inline test/unit/orchestration/dhcp_test.rb]: good catch

**Comment by dLobatog** [comment]
Merged as 85c6f530855ffafbb90c02e1f9dadabe1b6823ac, thanks @ares!

**Pull request closed without merging**